Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -185,9 +186,30 @@ public IPooledConnection Acquire()
 
                 if (!_availableConnections.TryTake(out connection))
                 {
-                    connection = CreateNewPooledConnection();
+                    var connAcquisitionTimer = new Stopwatch();
```

Comment:
We may come up with a solution by incorporating CancellationTokens inside this code and make the timeout value to be applied across recursive calls. This behaviour makes more sense to me, what about you? If you agree, I can work on another PR after this one's approval.

Reply:
Do we need to make it recursive?
I thought`acquisition timeout` measures the waiting time to acquire a connection from pool each time. If we enter some recursive call at line 215, then it is another new timeout?

Reply:
From user's point of view, if I set a connection acquisition timeout value I want it to be applied to the whole process whatever it does inside. At least I think that way :)

Reply:
I will leave the final decision of this setting to you and @lutovich and/or anyone else who might be interested.

Reply:
I initially thought about connection acquisition timeout as being a per-attempt thingy. However it probably makes more sense to have it per the whole `Pool#acquire()` call. It is a user-facing setting and users can only reason about total time it takes to acquire a connection from the pool.